src/node/desktop/src/main/args-manager.ts

```diff
@@ -0,0 +1,104 @@
+import { app } from 'electron';
+import { WinstonLogger } from '../core/winston-logger';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+
+import { enableDiagnosticsOutput, parseCommandLineLogLevel, setLogger } from '../core/logger';
+import { Application } from './application';
+import { exitSuccess, ProgramStatus, run } from './program-status';
+import { getComponentVersions, userLogPath } from './utils';
+
+// How to use:
+// [MAC] RStudio.app/Contents/MacOS/RStudio --session-exit
+// [MAC] RStudio.app/Contents/MacOS/RStudio --help
+// npm run start -- -- --session-exit
+// npm run start -- -- --log-level=err
+// npm run start -- -- --help
+// npm run start -- -- --version
+// RStudio command-line switches
+export const kRunDiagnosticsOption = '--run-diagnostics';
+export const kVersion = '--version';
+export const kVersionJson = '--version-json';
+export const kLogLevel = 'log-level';
+export const kDelaySession = 'session-delay';
+export const kSessionExit = 'session-exit';
+
+// RStudio Pro Only
+// export const kSessionServerOption = '--session-server';
+// export const kSessionServerUrlOption = '--session-url';
+// export const kTempCookiesOption = '--use-temp-cookies';
+
+// !IMPORTANT: If some args should early exit the application, add them to `webpack.plugins.js`
+export class ArgsManager {
+  argsList = [
+    {
+      name: kVersionJson,
+      describe: 'Report an extended version info',
```

Comment:
`Display versions of major components in JSON format`

Reply:
Fixed!